## Тема: Случайные события. Условная вероятность. Формула Байеса. Независимые испытания.

In [1]:
from math import factorial

# функция расчета количества сочетаний
def combinations(k, n):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

### 1. Из колоды в 52 карты извлекаются случайным образом 4 карты.  

### a) Найти вероятность того, что все карты – крести.

$$P(А) = \frac{m - число \;исходов, благоприятствующих \;нашему \;событию}{n - общее \;число \;исходов}$$  

**m** - количество **сочетаний** 4 карт крести из 52.  
$m = C_{13}^4$

In [7]:
m = combinations(4, 52/4)
m

715

**n** - все возможные сочетания 4 карт из 52  
$n = C_{52}^4$

In [8]:
n = combinations(4, 52)
n

270725

In [11]:
P = (m / n)
P

0.0026410564225690276

**Ответ:** вероятность взять 4 карты крести из 52 карт - 0,0026 или около 0.3%

### б) Найти вероятность, что среди 4-х карт окажется ХОТЯ БЫ один туз.

**m** - сумма **сочетаний** 1, 2, 3, 4 тузов из 52 карт  
m = A1 + A2 + A3 + A4   
A1 = 4 карты содержит 1 туз = $C_{4}^1 * C_{48}^3$   
А2 = 4 карты содержит 2 туза = $C_{4}^2 * C_{48}^2$    
А3 = 4 карты содержит 3 туза = $C_{4}^3 * C_{48}^1$   
А4 = 4 карты содержит 4 туза = $C_{4}^4 * C_{48}^0$   

In [40]:
A1 = combinations(1, 4) * combinations(3, 48)
A1

69184

In [41]:
A2 = combinations(2, 4) * combinations(2, 48)
A2

6768

In [42]:
A3 = combinations(3, 4) * combinations(1, 48)
A3

192

In [43]:
A4 = combinations(4, 4) * combinations(0, 48)
A4

1

In [44]:
m = A1 + A2 + A3 + A4
m

76145

In [45]:
P = (m / n)
P

0.2812632745405855

**Ответ:** вероятность взять 4 с хотя бы одним тузом из 52 карт - 0,2812 или около 28%

### 2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Так как задействованы не все кнопки и порядок не важен можно использовать формулу **сочетаний**.

$$C_n^k = \frac{n!}{k!(n - k)!}$$

**n** = 10 цифр  
**k** = 3 цифры, которые нужно нажать одновременно.

$$C_n^k = \frac{10!}{3!(10-3)!} = \frac{8 * 9 * 10}{2 * 3} = 120$$

In [33]:
C = combinations(3, 10)
C

120

In [46]:
P = 1 / C
P

0.008333333333333333

**Ответ:** вероятность того, что человек, не знающий код, откроет дверь с первой попытки 0.0083 или 0.8%

### 3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

**m** - число сочетаний извлечения 3 деталей из 9 окрашенных  
**n** - число сочетаний извлечения 3 деталей из 15

In [46]:
m = combinations(3, 9)
m

84

In [47]:
n = combinations(3, 15)
n

455

По формуле **классической вероятности** вычислим вероятность того, что 3 извлеченные детали из 15 являются окрашенными.

In [48]:
P = m / n
P

0.18461538461538463

**Ответ:** вероятность того, что 3 извлеченные детали из 15 окрашены 0.1846 или 18%

### 4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

Так как одно успешное событие не имеет значения без другого, то эти события являются **зависимыми**.

$$P(A*B) = P(A) * P(B|A)$$

**P(A) - m**(число исходов, благоприятствующих первому событию) / **n**(общее число исходов)  

m = 2, n = 100

In [6]:
A = 2 / 100
A

0.02

**P(B|A) - m**(число исходов, благоприятствующих второму событию) / **n**(общее число исходов минус число исходов, благоприятствующих первому событию) 

In [7]:
B_A = 1 / (100 - 1)
B_A

0.010101010101010102

In [8]:
P = A * B_A
P

0.00020202020202020205

**Ответ:** вероятность того, что 2 приобретенных билета окажутся выигрышными - 0.0002 или 0.02%